contracts/strategies/88mph/base/EightyEightMphStrategy.sol

```diff
@@ -0,0 +1,676 @@
+            uint64 depositMaturity;
+            (virtualTokenTotalSupply,,,, depositMaturity,) = IDInterest(rewardPool()).getDeposit(depositId);
+
+            // check if we reached maturity
+            bool early = block.timestamp <= depositMaturity;
+
+            uint256 withdrawVirtualAmount = amount;
+
+            if(early) {
+                // before maturation, virtualTokenAmount passed in to IDInterest.withdraw does not match underlying 1:1
+                // to get the ratio of underlying to virtualToken, we can use virtualTokenTotalSupply to rewardPoolBalance()
+                withdrawVirtualAmount = needToWithdraw.mul(virtualTokenTotalSupply).div(rewardPoolBalance());
+            } 
+            // after maturation virtualAmount = 1:1 with underlying, so no else statement needed here
+
+            uint256 toWithdraw = Math.min(virtualTokenTotalSupply, withdrawVirtualAmount);
+
+            IDInterest(rewardPool()).withdraw(depositId, toWithdraw, early);
+        }
+
+        IERC20(underlying()).safeTransfer(vault(), amount);
+    }
+
+    /**
+     * @dev Note that we currently don't have a mechanism here to include
+     *      the accrued rewards.
+     */
+    function investedUnderlyingBalance() external view returns (uint256) {
+        if (address(rewardPool()) == address(0)) {
+            return IERC20(underlying()).balanceOf(address(this));
+        }
+        // Add the amount locked in the reward pool and the amount that resides in this contract
+        // (both are in the units of "underlying")
+        // The second part is needed because of the emergency exit mechanism, which would break
+        // the assumption that all funds are inside the reward pool
+        return
+            rewardPoolBalance().add(
+                IERC20(underlying()).balanceOf(address(this))
+            );
+    }
+
+    /**
+     * @dev Governance or controller can claim coins that are somehow transferred
+     *      into the contract (eg. by mistake). Note that the underlying LP token
+     *      is not salvagable.
+     */
+    function salvage(
+        address recipient,
+        address token,
+        uint256 amount
+    ) external onlyControllerOrGovernance {
+        // Make sure governance or controller cannot come in and take away the invested tokens
+        require(
+            !unsalvagableTokens(token),
+            "Token is defined as non-salvagable"
+        );
+        IERC20(token).safeTransfer(recipient, amount);
+    }
+
+    function depositArbCheck() public view returns (bool) {
+        return true;
+    }
+
+    function doHardWork() external onlyNotPausedInvesting restricted {
+        // fail as early as possible if deposit must be rolled over
+        uint64 depositMaturity = depositMaturity();
+        if(depositMaturity != 0 && block.timestamp > depositMaturity) {
+            // time to roll over the deposit to a new depositId with extended maturity
+            if(!shouldRolloverDeposit) {
```

Comment:
rollingOver a deposit to a new maturation target can be triggered in two ways:
 - either through a doHardWork, **IF** the shouldRolloverDeposit has been set manually
 - or by alternatively running `rolloverDeposit`
 
 Doing this through doHardWork involves the vault which might hold underlying assets that are transfered to the strategy before running the rolloverDeposit. Depending on circumstances this might be useful.